Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,83 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
+                }
 
+                if (subjectEndpointWithValue.Value == null)
+                {
                     throw new EdFiSecurityException(
-                        $"Unable to authorize the request because there is no authorization support for associating the "
-                        + $"API client's associated education organization types ('{edOrgTypes}') with the resource.");
+                        $"Access to the resource item could not be authorized because the '{subjectEndpointWithValue.Name}' of the resource is empty.");
                 }
 
-                // Combine multiple statements resulting from multiple EdOrg types in the API client's claims using "OR"
-                // (This forces at least 1 of the relationships with the EdOrg types to exist)
-                segmentStatements.Add(string.Join($"{Environment.NewLine}OR ", segmentExpressions));
+                // Perform defensive checks against the remote possibility of SQL injection attack
+                ValidateTableNameParts(claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
+
+                string derivedAuthorizationViewName = ViewNameHelper.GetFullyQualifiedAuthorizationViewName(
+                    subjectEndpointName,
+                    claimEndpointName,
+                    authorizationSegment.AuthorizationPathModifier);
+
+                if (!IsAuthorizationViewSupported(derivedAuthorizationViewName))
+                {
+                    unsupportedAuthorizationViews.Add(derivedAuthorizationViewName);
+
+                    edOrgTypes.AddRange(authorizationSegment.ClaimsEndpoints
+                       .Select(s => s.Name.TrimSuffix("Id")));                    
+
+                    continue;
+                }
+
+                segmentStatements.Add(CreateSegmentExpression(ref parameterIndex));
+
+                string CreateSegmentExpression(ref int index)
+                {
+                    return string.Format(
+                        StatementTemplate,
+                        GetMultiValueCriteriaExpression(authorizationSegment.ClaimsEndpoints.ToList(), parameters, ref index),
+                        subjectEndpointName.ContainsIgnoreCase("EducationOrganizationId")
+                        ? "TargetEducationOrganizationId" : subjectEndpointName,
+                        GetSingleValueCriteriaExpression(subjectEndpointWithValue, parameters, ref index),
+                        derivedAuthorizationViewName);
+                }
+            }
+
+            if (unsupportedAuthorizationViews.Any())
+            {
+                _logger.Debug("Unable to authorize resource item because none of the following authorization views exist: "
+                    + $"'{string.Join("', '", unsupportedAuthorizationViews)}'");
+
+                var edOrgTypeValues = string.Join("', '", edOrgTypes.Distinct().OrderBy(x => x));
+
+                throw new EdFiSecurityException(
```

Comment:
This error is based on the idea that if a view doesn't exist for a concrete-to-concrete authorization segment that the authorization isn't supported. This is no longer the case. The only time that this condition will hit is if there is an authorization path modifier that comes from the security metadata and there is no view that supports that. This condition no longer has anything to do with the EdOrg types in the claims (technically, all EdOrg sub types are expected to be supported, but this depends on the data written into the EdOrgIdToEdOrgId table) -- it's now entirely about the authorization segment.

Change the error message to reflect that reality with this new view strategy.

Reply:
The error can still happen, but it will be because of the authorization path modifier... not the EdOrg types in the claims.

Reply:
I think this error message is now too specific with regards to authorization path modifiers. It also doesn't cover any scenario related to the expansion of Student/Staff/Parent USI (which is the other thing that could change in the system to possibly cause this failure).

Can you remove the identification of the authorization path modifiers specifically, and just list the authorization view(s) that were not found but were expected to exist and leave it at that?